src/main/java/com/conveyal/osmlib/OSM.java

```diff
@@ -181,6 +181,7 @@ public void readFromFile(String filePath) {
                 // FIXME this takes two minutes on NL OSM. We should probably save the intersections in a MapDB table.
                 LOG.info("Detecting intersections...");
                 for (Way way : ways.values()) {
+                    if (way.hasTag("building")) continue;
```

Comment:
This patch is only intended to fix intersection detection, so we've decided that skipping buildings entirely (if we ever do it) will be in another pull request.